src/garage/_functions.py

```diff
@@ -1,10 +1,71 @@
 """Functions exposed directly in the garage namespace."""
+from collections import defaultdict
+
 from dowel import tabular
 import numpy as np
 
+import garage
 from garage.misc.tensor_utils import discount_cumsum
 
 
+def log_multitask_performance(itr, batch, discount, task_names=None):
+    """Log per-task performance and average performance.
+
+    Args:
+        itr (int): Iteration.
+        batch (TrajectoryBatch): The trajectories to evaluate with.
+        discount (float): Discount value, from algorithm's property.
+        task_names (dict[str]): Names of all tasks to evaluate.
+            In case when a task in task_names does not exist in the give batch,
+            log NaN as a placeholder.
+
+    Returns:
+        numpy.ndarray: Undiscounted returns.
+
+    """
+    traj_by_name = defaultdict(list)
+    for trajectory in batch.split():
+        try:
+            task_name = trajectory.env_infos['task_name'][0]
```

Comment:
can you leave an issue about unifying this interface?

Reply:
Opened at https://github.com/rlworkgroup/garage/issues/1198